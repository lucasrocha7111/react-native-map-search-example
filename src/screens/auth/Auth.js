import React from 'react'
import {
    View,
    Image,
    TextInput,
    TouchableOpacity,
    Text,
    ActivityIndicator
} from 'react-native'
import * as Facebook from 'expo-facebook'
import { FACEBOOK_APP_ID } from '../../utils/constants'
import Styles, { Colors } from '../../themes/default/styles'

export class Auth extends React.Component {

    state = {
        isLoading: false
    }
    emailInput = null

    render() {
        return (
            <View style={[{flex: 1, justifyContent: 'center', width: '100%'}, Styles.backgroundStyle]}>
                <View style={[Styles.container, Styles.center]}>
                    <View style={[Styles.center]}>
                        <Image source={require('../../../assets/img/invillia-logo.png')} style={[{width: 100, height: 100, marginBottom: 40}]} />
                    </View>
                    <View style={[Styles.center, {justifyContent: 'center', width: '100%'}]}>
                        <TouchableOpacity style={[Styles.regularFont, Styles.btnDefault, {width: '100%', backgroundColor: Colors.blue}]} onPress={() => {
                            this._facebookLogin()
                        }}>
                            <Text style={[Styles.regularFont, {color: Colors.white}]}>Entrar com Facebook</Text>
                        </TouchableOpacity>
                        <TextInput ref={(c) => { this.emailInput = c }} style={[Styles.formInput]} placeholder={'Email'} />
                        <TextInput style={[Styles.formInput]} placeholder={'Senha'} />
                        <TouchableOpacity style={[Styles.btnDefault, Styles.center, {width: '100%'}]} onPress={() => {
                            this._login()
                        }}>
                            {this.state.isLoading
                            ? <ActivityIndicator />
                            : <Text style={[Styles.regularFont, Styles.center, {color: Colors.darkGrey, textAlign: 'center'}]}>Entrar</Text>
                            }
                        </TouchableOpacity>
                        
                    </View>
                </View>
            </View>
        )
    }

    _login = async () => {
        this.setState({
            isLoading: true
        })
        await setTimeout(() => {
            // fake request
            this.setState({
                isLoading: true
            })
            this.props.navigation.push('home')
        }, 1000)
    }

    _facebookLogin = async () => {
        try {
            let { type, token } = await Facebook.logInWithReadPermissionsAsync(FACEBOOK_APP_ID, {
                permissions: ['public_profile'],
            })
            console.log('data'. type, token)
            this.props.navigation.push('home')
        } catch(err) {
            console.log('Facebook error ', err)
        }
    }

}