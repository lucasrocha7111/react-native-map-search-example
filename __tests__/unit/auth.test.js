import React from 'react'
import renderer from 'react-test-renderer'
import { Auth } from '../../src/screens/auth/Auth'


describe('<Auth />', () => {
    it('has 1 child', () => {
        const tree = renderer.create(<Auth />).toJSON();
        expect(tree.children.length).toBe(1);
    })
    it('renders correctly', () => {
        const tree = renderer.create(<Auth />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})