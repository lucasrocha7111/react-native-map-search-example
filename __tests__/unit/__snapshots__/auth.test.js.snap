// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      },
      Object {
        "backgroundColor": "#f1f1f1",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "paddingHorizontal": 20,
          "width": "100%",
        },
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "alignSelf": "center",
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "height": 100,
              "marginBottom": 40,
              "width": 100,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "alignSelf": "center",
          },
          Object {
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4267b2",
            "borderColor": "#676e7d",
            "borderRadius": 5,
            "borderWidth": 0.8,
            "color": undefined,
            "fontFamily": "lato",
            "marginBottom": 20,
            "opacity": 1,
            "padding": 14,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": undefined,
                "fontFamily": "lato",
              },
              Object {
                "color": "#fff",
              },
            ]
          }
        >
          Entrar com Facebook
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        placeholder="Email"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "borderColor": "#676e7d",
              "borderRadius": 5,
              "borderWidth": 0.7,
              "color": "#676e7d",
              "fontFamily": "lato",
              "fontSize": 14,
              "height": 38,
              "marginBottom": 20,
              "paddingHorizontal": 10,
              "width": "100%",
            },
          ]
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        placeholder="Senha"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "borderColor": "#676e7d",
              "borderRadius": 5,
              "borderWidth": 0.7,
              "color": "#676e7d",
              "fontFamily": "lato",
              "fontSize": 14,
              "height": 38,
              "marginBottom": 20,
              "paddingHorizontal": 10,
              "width": "100%",
            },
          ]
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "alignSelf": "center",
            "borderColor": "#676e7d",
            "borderRadius": 5,
            "borderWidth": 0.8,
            "marginBottom": 20,
            "opacity": 1,
            "padding": 14,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": undefined,
                "fontFamily": "lato",
              },
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "alignSelf": "center",
              },
              Object {
                "color": "#676e7d",
                "textAlign": "center",
              },
            ]
          }
        >
          Entrar
        </Text>
      </View>
    </View>
  </View>
</View>
`;
